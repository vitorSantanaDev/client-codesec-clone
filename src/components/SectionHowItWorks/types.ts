import { ISectionHowItWorksProps } from 'interfaces/sections'

export type IViewProps = ISectionHowItWorksProps
