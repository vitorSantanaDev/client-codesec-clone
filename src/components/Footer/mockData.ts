export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
