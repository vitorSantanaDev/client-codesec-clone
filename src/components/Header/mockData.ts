import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
