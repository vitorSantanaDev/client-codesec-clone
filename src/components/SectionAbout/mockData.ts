export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    }
  ],
  illustration: {
    urlImage: '/images/about-ilustration.svg',
    altText: 'Illustration about'
  }
}
