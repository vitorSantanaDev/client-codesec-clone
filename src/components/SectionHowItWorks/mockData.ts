export const mockDataSectionProps = {
  title: 'Title Section example',
  subtitle: 'Subtitle',
  description:
    'Lorem ipsum dolor sit amet pretium sed, ullamcorper vel nisi. Cras varius ex sed dapibus sagittis. Cras orci libero, vestibulum tristique enim vel, volutpat euismod ligula.',
  steps: [
    {
      _id: String(Math.random() * 1000),
      icon: {
        altText: 'Icon',
        urlImage:
          'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg'
      },
      titleStep: 'Passo 1',
      descriptionStep:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent leo urna, semper vel pretium sed, ullamcorper vel nisi. Cras varius ex sed dapibus sagittis. Cras orci libero, vestibulum tristique enim vel, volutpat euismod ligula.'
    },
    {
      _id: String(Math.random() * 1000),
      icon: {
        altText: 'Icon',
        urlImage:
          'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg'
      },
      titleStep: 'Passo 1',
      descriptionStep:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent leo urna, semper vel pretium sed, ullamcorper vel nisi. Cras varius ex sed dapibus sagittis. Cras orci libero, vestibulum tristique enim vel, volutpat euismod ligula.'
    },
    {
      _id: String(Math.random() * 1000),
      icon: {
        altText: 'Icon',
        urlImage:
          'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg'
      },
      titleStep: 'Passo 1',
      descriptionStep:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent leo urna, semper vel pretium sed, ullamcorper vel nisi. Cras varius ex sed dapibus sagittis. Cras orci libero, vestibulum tristique enim vel, volutpat euismod ligula.'
    }
  ]
}
