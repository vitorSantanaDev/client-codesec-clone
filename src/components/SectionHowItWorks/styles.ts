import styled, { css } from 'styled-components'
import { Wrapper as Heading } from 'components/Heading/styles'
import { Wrapper as Subtitle } from 'components/Subtitle/styles'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.xhuge} 0;
  `}
`

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    > ${Subtitle} {
      margin-bottom: calc(${theme.spacings.small} - 0.8rem);
    }

    > ${Heading} {
      width: 100%;
      max-width: 592px;

      @media screen and (max-width: 520px) {
        margin-bottom: ${theme.spacings.medium};
      }

      @media ${theme.media.lteMedium} {
        font-size: calc(${theme.font.sizes.large} + 0.4rem);
      }
    }
  `}
`

export const AnimateWrapper = styled.div``

export const TitleAndDescriptionWrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: grid;
    align-items: baseline;
    grid-template-columns: 1fr 400px;
    margin-bottom: ${theme.spacings.huge};

    @media screen and (max-width: 520px) {
      justify-content: center;
      grid-template-columns: 1fr;
    }
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    width: 100%;
    max-width: 404px;
    line-height: 150%;
    color: ${theme.colors.gray7};
    font-size: ${theme.font.sizes.small};
  `}
`

export const BoxsWrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    align-items: center;
    gap: ${theme.spacings.medium};
    justify-content: space-between;

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      justify-content: center;
    }
  `}
`

export const DescriptionBoxStep = styled.p`
  ${({ theme }) => css`
    width: 100%;
    max-width: 300px;
    line-height: 150%;
    color: ${theme.colors.gray7};
    font-size: ${theme.font.sizes.small};
  `}
`

export const BoxStep = styled.div`
  ${({ theme }) => css`
    width: 100%;
    position: relative;
    padding: ${theme.spacings.medium} ${theme.spacings.small};
    background: ${theme.colors.blackDark};
    margin-bottom: ${theme.spacings.medium};

    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;

    border-radius: 3px;
    transition: 300ms ease-in-out;

    &::before {
      content: '';
      position: absolute;
      inset: 0;
      z-index: -1;
      margin: -1.5px;
      border-radius: 3px;
      background: ${theme.colors.redGradientDark};
    }

    &:hover {
      background: ${theme.colors.redGradientLight};
    }

    > ${Heading} {
      position: relative;
      font-size: ${theme.font.sizes.medium};
      margin-bottom: calc(${theme.spacings.medium} - 0.4rem);

      /* &::before {
        content: '';
        width: 345px;
        height: 1px;
        top: -51px;
        left: -144px;
        position: absolute;
        display: inline-block;
        background: ${theme.colors.redGradientDark};
      } */
    }
  `}
`

export const Icon = styled.img`
  ${({ theme }) => css`
    width: 64px;
    z-index: 10000;
    margin-bottom: calc(${theme.spacings.medium} - 0.4rem);
  `}
`
