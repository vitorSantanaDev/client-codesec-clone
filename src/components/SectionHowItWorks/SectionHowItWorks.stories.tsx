import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionHowItWorks from '.'
import { mockDataSectionProps } from './mockData'

export default {
  title: 'SectionHowItWorks',
  component: SectionHowItWorks,
  args: {
    ...mockDataSectionProps
  }
} as ComponentMeta<typeof SectionHowItWorks>

export const Default: ComponentStory<typeof SectionHowItWorks> = (args) => (
  <SectionHowItWorks {...args} />
)
