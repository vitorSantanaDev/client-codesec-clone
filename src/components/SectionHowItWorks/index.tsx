import Container from 'components/Container'
import Heading from 'components/Heading'
import Subtitle from 'components/Subtitle'

import { IViewProps } from './types'
import * as S from './styles'

const SectionHowItWorks = ({
  steps,
  title,
  subtitle,
  description
}: IViewProps) => (
  <Container>
    <S.Wrapper>
      <S.TitleAndDescriptionWrapper>
        <S.ContentWrapper data-aos="fade-right">
          <Subtitle>{subtitle}</Subtitle>
          <Heading>{title}</Heading>
        </S.ContentWrapper>
        <S.Description data-aos="fade-right">{description}</S.Description>
      </S.TitleAndDescriptionWrapper>
      <S.BoxsWrapper>
        {steps.map((step) => {
          return (
            <S.AnimateWrapper data-aos="fade-up-right" key={step._id}>
              <S.BoxStep>
                <S.Icon src={step.icon.urlImage} alt={step.icon.altText} />
                <Heading as="h5" size="medium">
                  {step.titleStep}
                </Heading>
                <S.DescriptionBoxStep>
                  {step.descriptionStep}
                </S.DescriptionBoxStep>
              </S.BoxStep>
            </S.AnimateWrapper>
          )
        })}
      </S.BoxsWrapper>
    </S.Wrapper>
  </Container>
)

export default SectionHowItWorks
