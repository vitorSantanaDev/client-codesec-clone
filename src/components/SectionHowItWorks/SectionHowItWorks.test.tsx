import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'
import SectionHowItWorks from '.'
import { mockDataSectionProps } from './mockData'

describe('<SectionHowItWorks />', () => {
  it('should render title, subtitle and description', () => {
    renderTheme(<SectionHowItWorks {...mockDataSectionProps} />)

    const subtitle = screen.getByText(/subtitle/i)
    const heading = screen.getByRole('heading', {
      name: /title section example/i
    })
    const description = screen.getByText(`${mockDataSectionProps.description}`)

    expect(subtitle).toBeInTheDocument()
    expect(heading).toBeInTheDocument()
    expect(description).toBeInTheDocument()
  })
  it('should render the correct amount of steps', () => {
    renderTheme(<SectionHowItWorks {...mockDataSectionProps} />)

    const titles = screen.getAllByRole('heading', { name: /passo 1/i })
    const icons = screen.getAllByRole('img', { name: /icon/i })

    expect(titles).toHaveLength(3)
    expect(icons).toHaveLength(3)
  })
  it('should render the correct amount of steps', () => {
    renderTheme(<SectionHowItWorks {...mockDataSectionProps} />)

    const titles = screen.getAllByRole('heading', { name: /passo 1/i })
    const icons = screen.getAllByRole('img', { name: /icon/i })

    expect(titles).toHaveLength(3)
    expect(icons).toHaveLength(3)
  })
  it('should render the correct amount of steps', () => {
    renderTheme(<SectionHowItWorks {...mockDataSectionProps} />)

    const titles = screen.getAllByRole('heading', { name: /passo 1/i })
    const icons = screen.getAllByRole('img', { name: /icon/i })

    expect(titles).toHaveLength(3)
    expect(icons).toHaveLength(3)
  })
  it('should match snapshots', () => {
    const { container } = renderTheme(
      <SectionHowItWorks {...mockDataSectionProps} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
