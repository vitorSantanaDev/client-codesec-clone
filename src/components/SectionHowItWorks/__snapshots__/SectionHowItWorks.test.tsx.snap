// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionHowItWorks /> should match snapshots 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  font-size: 5.6rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

.c12 {
  font-size: 2.4rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

.c5 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem - 0.2rem);
}

.c1 {
  width: 100%;
  padding: 6.4rem 0;
}

.c3 > .c4 {
  margin-bottom: calc(1.6rem - 0.8rem);
}

.c3 > .c6 {
  width: 100%;
  max-width: 592px;
}

.c2 {
  width: 100%;
  display: grid;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  grid-template-columns: 1fr 400px;
  margin-bottom: 5.6rem;
}

.c8 {
  width: 100%;
  max-width: 404px;
  line-height: 150%;
  color: #A3A3A3;
  font-size: 1.6rem;
}

.c9 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2.4rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c13 {
  width: 100%;
  max-width: 300px;
  line-height: 150%;
  color: #A3A3A3;
  font-size: 1.6rem;
}

.c10 {
  width: 100%;
  position: relative;
  padding: 2.4rem 1.6rem;
  background: #0A0909;
  margin-bottom: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 3px;
  -webkit-transition: 300ms ease-in-out;
  transition: 300ms ease-in-out;
}

.c10::before {
  content: '';
  position: absolute;
  inset: 0;
  z-index: -1;
  margin: -1.5px;
  border-radius: 3px;
  background: linear-gradient(225.05deg,#FA5770 0%,rgba(232,55,82,0) 100%);
}

.c10:hover {
  background: linear-gradient(225.05deg,rgba(250,87,112,0.1) 0%,rgba(232,55,82,0) 50%);
}

.c10 > .c6 {
  position: relative;
  font-size: 2.4rem;
  margin-bottom: calc(2.4rem - 0.4rem);
}

.c11 {
  width: 64px;
  z-index: 10000;
  margin-bottom: calc(2.4rem - 0.4rem);
}

@media (max-width:768px) {
  .c7 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c12 {
    font-size: 4.0rem;
  }
}

@media screen and (max-width:520px) {
  .c3 > .c6 {
    margin-bottom: 2.4rem;
  }
}

@media (max-width:768px) {
  .c3 > .c6 {
    font-size: calc(3.2rem + 0.4rem);
  }
}

@media screen and (max-width:520px) {
  .c2 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c9 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        data-aos="fade-right"
      >
        <span
          class="c4 c5"
          color="redGradient"
        >
          Subtitle
        </span>
        <h1
          class="c6 c7"
          color="white"
        >
          Title Section example
        </h1>
      </div>
      <p
        class="c8"
        data-aos="fade-right"
      >
        Lorem ipsum dolor sit amet pretium sed, ullamcorper vel nisi. Cras varius ex sed dapibus sagittis. Cras orci libero, vestibulum tristique enim vel, volutpat euismod ligula.
      </p>
    </div>
    <div
      class="c9"
    >
      <div
        class=""
        data-aos="fade-up-right"
      >
        <div
          class="c10"
        >
          <img
            alt="Icon"
            class="c11"
            src="https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg"
          />
          <h5
            class="c6 c12"
            color="white"
          >
            Passo 1
          </h5>
          <p
            class="c13"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent leo urna, semper vel pretium sed, ullamcorper vel nisi. Cras varius ex sed dapibus sagittis. Cras orci libero, vestibulum tristique enim vel, volutpat euismod ligula.
          </p>
        </div>
      </div>
      <div
        class=""
        data-aos="fade-up-right"
      >
        <div
          class="c10"
        >
          <img
            alt="Icon"
            class="c11"
            src="https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg"
          />
          <h5
            class="c6 c12"
            color="white"
          >
            Passo 1
          </h5>
          <p
            class="c13"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent leo urna, semper vel pretium sed, ullamcorper vel nisi. Cras varius ex sed dapibus sagittis. Cras orci libero, vestibulum tristique enim vel, volutpat euismod ligula.
          </p>
        </div>
      </div>
      <div
        class=""
        data-aos="fade-up-right"
      >
        <div
          class="c10"
        >
          <img
            alt="Icon"
            class="c11"
            src="https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg"
          />
          <h5
            class="c6 c12"
            color="white"
          >
            Passo 1
          </h5>
          <p
            class="c13"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent leo urna, semper vel pretium sed, ullamcorper vel nisi. Cras varius ex sed dapibus sagittis. Cras orci libero, vestibulum tristique enim vel, volutpat euismod ligula.
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`
